File: 1424-diagonal-traverse-ii/1424-diagonal-traverse-ii.java
class Solution {
    public int[] findDiagonalOrder(List<List<Integer>> nums) {
        List<List<Integer>> temp = new ArrayList<>();
        int count = 0;
        for(int i=0; i<nums.size(); i++){
            int len = nums.get(i).size();                        
            while((len+i)-temp.size() > 0){
                temp.add(new ArrayList<>());
            }
            int index = i;
            for(int j=0; j<len; j++){
                temp.get(index++).add(nums.get(i).get(j));
                
            }
            count += len;
        }
        int index = 0;
        int[] res = new int[count];
        for(List<Integer> row : temp){
            for(int i=row.size()-1; i>=0; i--){
                res[index++] = row.get(i);
            }
        }
        return res;
    }
}

File: 1424-diagonal-traverse-ii/1424-diagonal-traverse-ii.cpp
class Solution {
public:
    vector<int> findDiagonalOrder(vector<vector<int>>& nums) {
        vector<std::vector<int>> temp;
    int count = 0;
    for (int i = 0; i < nums.size(); i++) {
        int len = nums[i].size();
        if (len + i > temp.size()) {
            while ((len + i) - temp.size() > 0) {
                temp.push_back(vector<int>());
            }
        }
        int index = i;
        for (int j = 0; j < len; j++) {
            temp[index++].push_back(nums[i][j]);
        }
        count += len;
    }
    int index = 0;
    vector<int> res(count);
    for (std::vector<int>& row : temp) {
        for (int i = row.size() - 1; i >= 0; i--) {
            res[index++] = row[i];
        }
    }
    return res;
    }
};